standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/PropertyServlet.java

```diff
@@ -331,42 +308,34 @@ private void runGet(HttpServletRequest request,
     }
   }
 
+  public static ServletServerBuilder.Descriptor createServlet(Configuration configuration) {
+    try {
+      int port = MetastoreConf.getIntVar(configuration, MetastoreConf.ConfVars.PROPERTIES_SERVLET_PORT);
+      String path = MetastoreConf.getVar(configuration, MetastoreConf.ConfVars.PROPERTIES_SERVLET_PATH);
+      if (port >= 0 && path != null && !path.isEmpty()) {
+        ServletSecurity security = new ServletSecurity(configuration);
+        HttpServlet servlet = security.proxy(new PropertyServlet(configuration));
+        return new ServletServerBuilder.Descriptor(port, path, servlet) {
+          @Override public String toString() {
+            return "HMS property";
```

Comment:
should we reuse https://github.com/apache/hive/pull/5606/files#diff-11d6b8263574d38f07d091c65a599f4b8ca9c1bea8fc221d7db0338c62e69624R69?
i.e. `servlet.getServletName()`

Reply:
getServletName() is only valid after calling init() on the servlet instance otherwise it throws an exception and I'd prefer the toString() method to be idempotent.

Reply:
you could override `getServletName()` in proxy and propagate to `delegate` instead of code duplication